datafusion/core/src/physical_plan/windows/mod.rs

```diff
@@ -187,6 +188,30 @@ fn create_built_in_window_expr(
     })
 }
 
+pub(crate) fn calc_requirements(
+    partition_by_exprs: &[Arc<dyn PhysicalExpr>],
+    orderby_sort_exprs: &[PhysicalSortExpr],
+) -> Option<Vec<PhysicalSortRequirement>> {
+    let mut sort_reqs = vec![];
+    for partition_by in partition_by_exprs {
+        sort_reqs.push(PhysicalSortRequirement {
+            expr: partition_by.clone(),
+            options: None,
```

Comment:
Is it correct that being able to express `options:None` for partitioning operations is the key rationale (thing we can't do before) for this PR?

Reply:
Right. I vaguely remember us discussing internally that there are also some other use cases where some sort of ordering is required but the options don't matter, but I can't recall now

Reply:
Another example that the sorting direction is not that important is the SortMergeJoin, but the additional requirements
is for all its input, the ordering should be aligned, should be `ASC` or `DESC` altogether. I think the current `PhysicalSortRequirement` API can not represent this alignment constraints explicitly.